app/decorators/base_decorator.rb

```diff
@@ -0,0 +1,17 @@
+class BaseDecorator
+  attr_reader :obj
+
+  def initialize(obj, &block)
+    @obj = obj
+
+    yield(self) if block_given?
+  end
+
+  def method_missing(method_name, *args)
+    obj.public_send(method_name, *args)
+  end
+
+  def respond_to_missing(method_name, include_private = false)
+    super
+  end
```

Comment:
We could also use `SimpleDecorator` which implements this for us. I believe it also tries to mitigate the performance penalty incurred by `method_missing` (eg. by defining methods lazily so that a second invocation meets a real method).